src/frontend/src/binder/expr/function.rs

```diff
@@ -166,7 +168,41 @@ impl Binder {
                 return Ok(ExprImpl::literal_varchar(self.db_name.clone()));
             }
             "current_schema" if inputs.is_empty() => {
-                return Ok(ExprImpl::literal_varchar(DEFAULT_SCHEMA_NAME.to_string()));
+                let first_schema_name = self
+                    .catalog
+                    .first_valid_schema(
+                        &self.db_name,
+                        &self.search_path,
+                        &self.auth_context.user_name,
+                    )
+                    .map(|schema| schema.name())
+                    .unwrap_or_default();
+                return Ok(ExprImpl::literal_varchar(first_schema_name));
+            }
+            "current_schemas" => {
+                if inputs.len() == 1 && let ExprImpl::Literal(literal) = &inputs[0] && literal.return_type() == DataType::Boolean {
+                    let mut schema_names = vec![];
+
+                    let v = literal.get_data().clone().unwrap().into_bool();
```

Comment:
Technically, the input parameter can be a non-literal (e.g. `current_schemas(v1)`). But I agree we only need to support literals here:
```
let ExprImpl::Literal(literal) = &inputs[0] else { return Err(unsupported); };
```

Even when it is literal, this `unwrap()` should be avoided because `current_schemas(null)` is `null` rather than panics.